umbra/umbra_64.c

```diff
@@ -1261,6 +1262,31 @@ umbra_insert_app_to_shadow_arch(void *drcontext,
                                 reg_id_t *scratch_regs,
                                 int num_scratch_regs)
 {
+#if defined(AARCH64)
+    reg_id_t tmp = *scratch_regs;
```

Comment:
You are relying on the user to reserve a scratch register; hence the dereference. Please add a check to ensure that the num_scratch_regs conforms to this assumption.

Also, fix umbra_num_scratch_regs_for_translation_arch() so that the user is aware that we require a scratch register.

Reply:
Okay thanks! Have just pushed my interpretation of what you have asked for.

Reply:
Sorry, I was not clear. Please have DRMF_ERROR_INVALID_PARAMETER be returned, instead of using an ASSERT.

Moreover, I believe the check needs to be >= and not >.

Reply:
ah okay yes, done